src/mon/MDSMonitor.cc

```diff
@@ -2385,16 +2390,14 @@ int MDSMonitor::legacy_filesystem_command(
   if (prefix == "mds set_max_mds") {
     // NOTE: deprecated by "fs set max_mds"
     int64_t maxmds;
-    if (!cmd_getval(g_ceph_context, cmdmap, "maxmds", maxmds) || maxmds < 0) {
+    if (!cmd_getval(g_ceph_context, cmdmap, "maxmds", maxmds) || maxmds <= 0) {
```

Comment:
I believe you need to be able to set maxmds 0 to do a proper shutdown and cleanup of the FS?